src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/DemultiplexingDispatchMessageFormatter.cs

```diff
@@ -0,0 +1,108 @@
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(formatters));
+            }
+
+            _formatters = new Dictionary<WebContentFormat, IDispatchMessageFormatter>();
+            foreach (WebContentFormat key in formatters.Keys)
+            {
+                _formatters.Add(key, formatters[key]);
+            }
+
+            _defaultFormatter = defaultFormatter;
+        }
+
+        public void DeserializeRequest(Message message, object[] parameters)
+        {
+            if (message == null)
+            {
+                return;
+            }
+
+            IDispatchMessageFormatter selectedFormatter;
+            if (TryGetEncodingFormat(message, out WebContentFormat format))
+            {
+                _formatters.TryGetValue(format, out selectedFormatter);
+                if (selectedFormatter == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.UnrecognizedHttpMessageFormat, format, GetSupportedFormats())));
+                }
+            }
+            else
+            {
+                selectedFormatter = _defaultFormatter;
+                if (selectedFormatter == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new InvalidOperationException(SR.Format(SR.MessageFormatPropertyNotFound3)));
+                }
+            }
+
+            selectedFormatter.DeserializeRequest(message, parameters);
+        }
+
+        public Message SerializeReply(MessageVersion messageVersion, object[] parameters, object result)
+        {
+            throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException(SR.Format(SR.SerializingReplyNotSupportedByFormatter, this)));
+        }
+
+        internal static string GetSupportedFormats(IEnumerable<WebContentFormat> formats)
+        {
+            StringBuilder sb = new StringBuilder();
+            int i = 0;
+            foreach (WebContentFormat format in formats)
+            {
+                if (i > 0)
+                {
+                    sb.Append(CultureInfo.CurrentCulture.TextInfo.ListSeparator);
+                    sb.Append(" ");
+                }
+                sb.Append("'" + format.ToString() + "'");
+                ++i;
+            }
+
+            return sb.ToString();
+        }
+
+        internal static bool TryGetEncodingFormat(Message message, out WebContentFormat format)
+        {
+            message.Properties.TryGetValue(WebBodyFormatMessageProperty.Name, out object prop);
+            if (!(prop is WebBodyFormatMessageProperty formatProperty))
+            {
+                format = WebContentFormat.Default;
+                return false;
+            }
+
+            format = formatProperty.Format;
+            return true;
+        }
+
+        string GetSupportedFormats()
+        {
```

Comment:
Accessibility modifier

Reply:
This is addressed in e575590ad880fad2afb332b5bf532abdcfbbd23b.